tutorials/how_to/span_representations.md

```diff
@@ -0,0 +1,180 @@
+* [Segmental Recurrent Neural Networks](https://arxiv.org/pdf/1511.06018.pdf)
+
+
+## Generating `SpanFields` from text in a `DatasetReader`
+
+`SpanFields` are a type of `Field` in AllenNLP which take a start index, an end index
+and a `SequenceField` which the indices refer to. Once a batch of `SpanFields` has been
+converted to a tensor, we will have a matrix of shape (batch_size, 2), where the last
+dimension contains the start and end indices passed in to the SpanField constructor.
+However, for many models, you'll want to represent _many_ spans for a single batch
+element - the way to do this is to use a `ListField[SpanFields]`, which will create
+a tensor of shape (batch_size, num_spans, 2) once indexed.
+
+
+## Extracting Span Representations from a text sequence
+
+In many cases, you will want to extract spans from vector representations of sentences.
+In order to do this in AllenNLP, you will need to use a `SpanExtractor`. Broadly, a 
+`SpanExtractor` takes a sequence tensor of shape `(batch_size, sentence_length, embedding_size)`
+and some indices of shape `(batch_size, num_spans, 2)` and return the relevant indices into
+the sequence as a tensor of shape `(batch_size, num_spans, encoded_size)`.
+
+The simplest `SpanExtractor` is the `EndpointSpanExtractor`, which epresents spans as a 
+combination of the embeddings of their endpoints.
+
+```
+import torch
+from torch.autograd import Variable
+from allennlp.modules.span_extractors import EndpointSpanExtractor
+sequence_tensor = Variable(torch.randn([2, 5, 7]))
+# concatentate start and end points together to form our representation.
+extractor = EndpointSpanExtractor(input_dim7, combination="x,y")
+
+# Typically these would come from a SpanField,
+# rather than being created directly.
+indices = Variable(torch.LongTensor([[[1, 3],
+                                      [2, 4]],
+                                     [[0, 2],
+                                      [3, 4]]]))
+
+# We concatenated the representations for the start and end of
+# the span, so the embedded span size is 2 * embedding_size.
+# shape (batch_size, num_spans, 2 * embedding_size).
+span_representations = extractor(sequence_tensor, indices)
+assert list(span_representations.size()) == [2, 2, 14]
+```
+
+There are other types of Span Extractors - for instance, the `SelfAttentiveSpanExtractor`,
+which computes span representations by generating an unnormalized attention score for each
+word in the sentence. Spans representations are then computed with respect to these
+scores by normalising the attention scores for words inside the span.
+
+
+## Scoring and Pruning Spans
+
+Span based representations have been effective for modelling/approximating structured
+prediction problems - however, many models which leverage this type of representation
+also involve some kind of _span enumeration_ (i.e considering all possible spans in a 
+sentence/document). For a given sentence of length n, there are n^2 spans. In itself, 
```

Comment:
Making this (and below) a [superscript](https://stackoverflow.com/a/15156585/1467533) would be nice.